src/test/monolith_opcodes_tests.cpp

```diff
@@ -139,14 +139,34 @@ static void RunTestForAllBitwiseOpcodesSizes(const valtype &a, const valtype &b,
                                              const valtype &expected_or,
                                              const valtype &expected_xor) {
     valtype wa, wb, wand, wor, wxor;
+    // Allocate once here, just to be sure.
+    wa.reserve(a.size());
+    wb.reserve(b.size());
+    wand.reserve(expected_and.size());
+    wor.reserve(expected_or.size());
+    wxor.reserve(expected_xor.size());
+    // Test for all len(wa) <= len(wb)
     for (size_t i = 0; i < a.size(); i++) {
         wa.push_back(a[i]);
         wb.push_back(b[i]);
         wand.push_back(expected_and[i]);
         wor.push_back(expected_or[i]);
         wxor.push_back(expected_xor[i]);
-
+        // Test for len(wa) == len(wb)
         RunTestForAllBitwiseOpcodes(wa, wb, wand, wor, wxor);
+        // Test for all len(wb) == len(wa)..len(b), expand arrays
+        for (size_t j = wa.size(); j < wb.size(); j++) {
```

Comment:
Something doesn't look right here, but I can't quite tell what's going on. `wb` is being resized while iterating on it?

Reply:
Shouldn't `wb.size()` and `wa.size()` be the same on the first pass here?

Reply:
Yes, basically we start with `{}` on both `wa` and `wb`, and check the AND/OR/XOR on that, then add one byte to `wb`, check the AND/OR/XOR again, then do the same for all bytes up to 520. Then we add one byte to `wa`, and check all strings `wb` that have equal or longer length again, rinse and repeat until we tested all lengths of `wa` up to 520.

Reply:
This code doesn't look like it does that to me. 

1. Start with {}
2. Push back to both `wa` and `wb` (Sizes are the same)
3. Loop for all sizes between `wa.size() and `wb.size() (This doesn't pass the check)
      3.a. Add bytes to `wb` which keeps increasing the size of `wb`
      
It looks like it'd be an infinite loop if it ever actually ran in the first place. Did I miss something?